trunner/runners/Stm32l4Runner.py

```diff
@@ -0,0 +1,77 @@
+#
+# Phoenix-RTOS test runner
+#
+# Stm32l4 runner
+#
+# Copyright 2021 Phoenix SYstems
+# Authors: Mateusz Niewiadomski
+#
+import subprocess
+import serial
+import time
+import pexpect
+import pexpect.fdpexpect
+
+from .common import DeviceRunner
+from .common import _BOOT_DIR
+
+
+class Stm32l4Runner(DeviceRunner):
+
+    class oneByOne_fdspawn(pexpect.fdpexpect.fdspawn):
+
+        def send(self, s):
+            for c in s:
+                super().send(c)
+                time.sleep(0.03)
+
+        def sendline(self, s):
+            self.send(s)
+            super().send('\n')
+
+    OPENOCD_CMD = ['openocd -f interface/stlink.cfg -f target/stm32l4x.cfg -c \
+"reset_config srst_only srst_nogate connect_assert_srst" -c "program ', ' 0x08000000 verify reset exit"']
+    IMAGE = 'phoenix-armv7m4-stm32l4x6.bin'
+
+    def __init__(self, port1, port2=None):
+        super().__init__(port1)
+        self.port2 = port2
+        self.image_flashed = False
+        self.openocd_cmd = self.OPENOCD_CMD[0] + str(_BOOT_DIR / self.IMAGE) + self.OPENOCD_CMD[1]
+
+    def flash(self):
+        # Flashing with openocd as a separate process
+        openocd_process = subprocess.Popen(
+            str(self.openocd_cmd),
+            shell=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE)
+
+        stdout, stderr = openocd_process.communicate()
+        openocd_process.wait()
+        # Chcecking openocd flashing result
+        verify1 = str(stderr).find('** Verified OK **')
+        verify2 = str(stderr).find('shutdown command invoked')
+        assert verify1 >= 0 and verify2 >= 0, "Stm32l4Runner: OpenOCD failed to flash target"
```

Comment:
I assume the `openocd` doesn't report the status by exit value (that would be better to check)

Usability note: maybe we should output logs from flashing for user to see (similarly to rt1064)?

Reply:
I miss the point that return value can be read from `subprocess.wait()`. 

Added that functionality. It works well, returning `!0` if no target present, no stlink present, or target present but flashing failed